src/lib/auth/safari-authenticator.js

```diff
@@ -0,0 +1,160 @@
+import Authenticator from '@/lib/auth/authenticator.js'
+
+/**
+ * This is a base class for all other authenticator classes.
+ * It defines an API that is common across all variations of authenticator objects.
+ * An authenticator object serves as a proxy between a component's Auth data module
+ * and platform-specific lower level authentication functionality.
+ */
+export default class SafariAuthenticator extends Authenticator {
+  /**
+   * @constructor
+   */
+  constructor () {
+    super()
+
+    // User data props
+    this.isAuthenticated = false
+    this.hasUserData = false
+    // A full name of the user
+    this.userName = null
+    // A user's nickname
+    this.userNickname = null
+    // A user ID
+    this.userId = null
+
+    this.accessToken = null
+
+    // TODO: Hardcoded for now. Can we do it any better than that?
+    this.endpoints = {
+      wordlist: 'https://userapis.alpheios.net/v1/words',
```

Comment:
can we get them from an env file? But I guess it's okay to leave them hard-coded for now, because unless we retrieve them from a service, a change requires a rebuild anyway.

Reply:
That were my thoughts exactly. We distribute Safari App Extension as a single piece app. All JS code is integrated into the `.app` file, the file is signed and maybe even encrypted and nothing can be changed there once the `.app` file is created. We cannot source any endpoint URL changes into it dynamically.

However, if endpoints changes, it might be convenient to source in the changes instead of manually searching and replacing lines in the code. Maybe it will be better to have endpoints in a separate file and import it from JS. What do you think?

Reply:
well, this is what we do for the chrome and safari extensions so, yes. Maybe we could just reuse that file? the env-webext.js that is in the dist directory?

Reply:
That would be a good solution, will implement